### PR TITLE
Adsk Contrib - Add operator<< for FormatMetadata

### PR DESCRIPTION
**Pull request by BernardLefebvre**
Add an operator<< for FormatMetadata so that it can be printed out (usefull in pyhton).
Adjust FormatMetadata::addChildElement so that it does not return.
Improve documentation.
Add python test for FormatMetadata.

Signed-off-by: Bernard Lefebvre <bernard.lefebvre@autodesk.com>

### CODE REVIEW

**Comment by hodoulp** [comment]
As usual this is a notice that we will merge this week (Thursday or Friday), if there are no discussions on the pull request.

File: src/bindings/python/PyFormatMetadata.cpp [hunk 1]
```diff
@@ -73,7 +81,11 @@ void bindPyFormatMetadata(py::module & m)
                 return ConstChildElementIterator(self);
             })
         .def("getChildElements", [](FormatMetadata & self) { return ChildElementIterator(self); })
-        .def("addChildElement", &FormatMetadata::addChildElement, "name"_a, "value"_a)
```
Comment (michdolan) [inline src/bindings/python/PyFormatMetadata.cpp:76]: I don't think this change is necessary. pybind11 handles `const char *` from Python string automatically. I have primarily been using `std::string` args in the constructors just for ease of checking for empty input.

Comment (BernardLefebvre) [reply src/bindings/python/PyFormatMetadata.cpp:76]: I'll revert that, but I also need to remove the test that check TypeError is raised when function is used with None for a parameter (segfault is generated)

Comment (michdolan) [reply src/bindings/python/PyFormatMetadata.cpp]: If you add `.none(false)` to the args which are causing segfaults when passing `None`, that should fix the issue. You can search for that in the bindings to see exmaples of where it has been added already.

**Review by michdolan — APPROVED** [review]

**Comment by michdolan** [comment]
Closing and reopening to trigger CI again, in case the error was VM-caused

**Review by hodoulp — APPROVED** [review]
